src/pymoca/ast.py

```diff
@@ -756,6 +833,76 @@ def _find_constant_symbol(self, component_ref: ComponentRef, search_parent=True)
     def find_constant_symbol(self, component_ref: ComponentRef) -> Symbol:
         return self._find_constant_symbol(component_ref)
 
+    def _find_symbol(self, component_ref: ComponentRef, search_parent=True, in_original_scope=False) -> Symbol:
+        if component_ref.child:
+            # Try classes first, and constant symbols second
+            t = component_ref.to_tuple()
+
+            try:
+                node = self._find_class(ComponentRef(name=t[0]), search_parent)
+                return node._find_symbol(ComponentRef.from_tuple(t[1:]), False)
+            except ClassNotFoundError:
+                try:
+                    s = self.symbols[t[0]]
+                except KeyError:
+                    raise SymbolNotFoundError()
+
+                # Found a symbol. Continue lookup on type of this symbol.
+                if isinstance(s.type, InstanceClass):
+                    return s.type._find_symbol(ComponentRef.from_tuple(t[1:]), False, in_original_scope)
+                elif isinstance(s.type, ComponentRef):
+                    node = self._find_class(s.type)  # Parent lookups is OK here.
```

Comment:
Same here, Probably faster to make this a copy=False lookup. That's also strictly speaking more correct.